python_modules/libraries/dagster-dbt/dagster_dbt/asset_defs.py

```diff
@@ -0,0 +1,153 @@
+    description_sections = [
+        node_info["description"],
+        "#### Columns:\n" + columns_to_markdown(node_info["columns"]),
+        f"#### Raw SQL:\n```\n{code_block}\n```",
+    ]
+    description = "\n\n".join(description_sections)
+
+    @asset(
+        name=node_info["name"],
+        description=description,
+        ins={
+            dep_name.split(".")[-1]: AssetIn(managed=False)
+            for dep_name in node_info["depends_on"]["nodes"]
+        },
+        required_resource_keys={"dbt"},
+        io_manager_key=io_manager_key,
+    )
+    def _node_asset(context):
+        context.resources.dbt.run(models=[".".join([node_info["package_name"], node_info["name"]])])
+        if runtime_metadata_fn:
+            metadata = runtime_metadata_fn(context, node_info)
+            yield Output(None, metadata=metadata)
+
+    return _node_asset
+
+
+def load_assets_from_dbt(
+    project_dir: Optional[str] = None,
+    profiles_dir: Optional[str] = None,
+    manifest_json: Optional[Mapping[str, Any]] = None,
+    manifest_path: Optional[str] = None,
+    select: Optional[str] = None,
+    runtime_metadata_fn: Optional[
+        Callable[[SolidExecutionContext, Mapping[str, Any]], Mapping[str, Any]]
+    ] = None,
+    io_manager_key: Optional[str] = None,
+) -> List[OpDefinition]:
+    """
+    Loads a set of DBT models into Dagster assets.
+
+    Creates one Dagster asset for each DBT model. You can point to dbt in one of few different ways:
+    by setting project_dir and profiles_dir, by pointing to a manifest.json file with manifest_path,
+    or by passing in a JSON-structured manifest blob into manifest_json.
+
+    Args:
+        project_dir (Optional[str]): The directory containing the DBT project to load.
+        profiles_dir (Optional[str]): The profiles directory to use for loading the DBT project.
+            Defaults to a directory called "config" inside the project_dir.
+        select (str): A DBT selection string for the models in a project that you want to include.
+            Defaults to "*".
+        manifest_json (Optional[Mapping[str, Any]]): The contents of a DBT manifest.json, which contains
+            a set of models to load into assets.
+        manifest_path (Optional[str]): A path to a DBT manifest.json, which contains a set of models
+            to load into assets.
+        runtime_metadata_fn: (Optional[Callable[[SolidExecutionContext, Mapping[str, Any]], Mapping[str, Any]]]):
+            A function that will be run after any of the assets are materialized and returns
+            metadata entries for the asset, to be displayed in the asset catalog for that run.
+        io_manager_key (Optional[str]): The IO manager key that will be set on each of the returned
+            assets. When other ops are downstream of the loaded assets, the IOManager specified
+            here determines how the inputs to those ops are loaded. Defaults to "io_manager_key".
```

Comment:
It seems like it actually just eventually defaults to "io_manager", because if you pass in None here, the key will remain None in _dbt_node_to_asset.

Reply:
Oops, that's what I meant